activejob/lib/active_job/queue_adapters/sidekiq_adapter.rb

```diff
@@ -18,70 +18,17 @@ module QueueAdapters
     #
     #   Rails.application.config.active_job.queue_adapter = :sidekiq
     class SidekiqAdapter < AbstractAdapter
-      def initialize(*) # :nodoc:
-        @stopping = false
+      @stopping = false
 
-        Sidekiq.configure_server do |config|
-          config.on(:quiet) { @stopping = true }
-        end
+      callback = -> { @stopping = true }
 
-        Sidekiq.configure_client do |config|
-          config.on(:quiet) { @stopping = true }
-        end
-      end
+      Sidekiq.configure_client { |config| config.on(:quiet, &callback) }
+      Sidekiq.configure_server { |config| config.on(:quiet, &callback) }
 
-      def enqueue(job) # :nodoc:
-        job.provider_job_id = JobWrapper.set(
-          wrapped: job.class,
-          queue: job.queue_name
-        ).perform_async(job.serialize)
-      end
+      delegate :enqueue, :enqueue_at, :enqueue_all, :stopping?,
+        :enqueue_after_transaction_commit?, to: Sidekiq::ActiveJob::QueueAdapters::SidekiqAdapter
 
-      def enqueue_at(job, timestamp) # :nodoc:
-        job.provider_job_id = JobWrapper.set(
-          wrapped: job.class,
-          queue: job.queue_name,
-        ).perform_at(timestamp, job.serialize)
-      end
-
-      def enqueue_all(jobs) # :nodoc:
-        enqueued_count = 0
-        jobs.group_by(&:class).each do |job_class, same_class_jobs|
-          same_class_jobs.group_by(&:queue_name).each do |queue, same_class_and_queue_jobs|
-            immediate_jobs, scheduled_jobs = same_class_and_queue_jobs.partition { |job| job.scheduled_at.nil? }
-
-            if immediate_jobs.any?
-              jids = Sidekiq::Client.push_bulk(
-                "class" => JobWrapper,
-                "wrapped" => job_class,
-                "queue" => queue,
-                "args" => immediate_jobs.map { |job| [job.serialize] },
-              )
-              enqueued_count += jids.compact.size
-            end
-
-            if scheduled_jobs.any?
-              jids = Sidekiq::Client.push_bulk(
-                "class" => JobWrapper,
-                "wrapped" => job_class,
-                "queue" => queue,
-                "args" => scheduled_jobs.map { |job| [job.serialize] },
-                "at" => scheduled_jobs.map { |job| job.scheduled_at&.to_f }
-              )
-              enqueued_count += jids.compact.size
-            end
-          end
-        end
-        enqueued_count
-      end
-
-      class JobWrapper # :nodoc:
-        include Sidekiq::Worker
-
-        def perform(job_data)
-          Base.execute job_data.merge("provider_job_id" => jid)
-        end
-      end
+      JobWrapper = Sidekiq::ActiveJob::Wrapper
```

Comment:
This constant was only just introduced in sidekiq/sidekiq#6430, which seems to have made it in sidekiq [v7.3.3](https://github.com/sidekiq/sidekiq/blob/main/Changes.md#733).

But Rails aims to maintain compatibility with older versions of Sidekiq, unless there is a good reason to break that guarantee and update the dependency requirement on the gem:
https://github.com/rails/rails/blob/5a7a59ab4a9ac34b8cadf9d8ce7b554da50c5e43/activejob/lib/active_job/queue_adapters/sidekiq_adapter.rb#L3

However, because we plan to deprecate and remove this adapter after giving users time to upgrade to a version of Sidekiq that contains the upstream adapter, this change is not acceptable.